src/main/java/org/openstreetmap/atlas/checks/database/FlagDatabaseSubCommand.java

```diff
@@ -0,0 +1,375 @@
+    private final OptionAndArgumentDelegate optionAndArgumentDelegate;
+    private Timestamp timestamp;
+
+    public static void main(final String[] args)
+    {
+        new FlagDatabaseSubCommand().runSubcommandAndExit(args);
+    }
+
+    public FlagDatabaseSubCommand()
+    {
+        this.optionAndArgumentDelegate = this.getOptionAndArgumentDelegate();
+    }
+
+    @Override
+    @SuppressWarnings("squid:S3655")
+    public int execute()
+    {
+        final Time timer = Time.now();
+        final DatabaseConnection database = new DatabaseConnection(
+                this.optionAndArgumentDelegate.getOptionArgument(DATABASE_URL_INPUT).get());
+        final Connection databaseConnection = database.getConnection();
+        final String inputPath = this.optionAndArgumentDelegate.getOptionArgument(FLAG_PATH_INPUT)
+                .get();
+        this.timestamp = new Timestamp(Instant.now().toEpochMilli());
+
+        new File(inputPath).listFilesRecursively().parallelStream().forEach(file ->
+        {
+            // If this file is something we handle, read and upload the tasks contained within
+            final Optional<OutputFileType> optionalHandledFileType = getOptionalOutputType(file);
+            optionalHandledFileType.ifPresent(outputFileType ->
+            {
+
+                try (BufferedReader reader = this.getReader(file, outputFileType))
+                {
+
+                    final PreparedStatement flagSqlStatement = databaseConnection.prepareStatement(
+                            "INSERT INTO flag(flag_id, check_name, instructions, date_created) VALUES (?,?,?,?);");
+
+                    final PreparedStatement featureSqlStatement = databaseConnection
+                            .prepareStatement(String.format(
+                                    "INSERT INTO feature (flag_id, geom, osm_id, atlas_id, iso_country_code, tags, item_type, date_created) VALUES (?,%s,?,?,?,?);",
+                                    "ST_GeomFromGeoJSON(?), ?, ?"));
+
+                    final List<String> lines = reader.lines().collect(Collectors.toList());
+                    int startIndex = 0;
+                    int endIndex;
+
+                    do
+                    {
+                        endIndex = Math.min(startIndex + BATCH_SIZE, lines.size());
+                        final List<String> batchLines = lines.subList(startIndex, endIndex);
+
+                        batchLines.forEach(line ->
+                        {
+                            final JsonObject parsedFlag = new JsonParser().parse(line)
+                                    .getAsJsonObject();
+                            final JsonArray features = this.filterOutPointsFromGeojson(
+                                    parsedFlag.get(FEATURES).getAsJsonArray());
+                            final Gson gson = new GsonBuilder().registerTypeAdapter(CheckFlag.class,
```

Comment:
I think you could pull this out into an instance or static variable so you don't have to initialize it every time you add a batch to the db

Reply:
yep, good idea